mshadow/tensor_gpu-inl.h

```diff
@@ -54,16 +54,17 @@ inline void FreeSpace(Tensor<gpu, dim, DType> *obj) {
   MSHADOW_CUDA_CALL(cudaFree(obj->dptr_));
   obj->dptr_ = NULL;
 }
-template<typename A, typename B, int dim, typename DType>
+template<typename A, typename B, int dim, typename DType, typename SrcDType>
```

Comment:
note this `CHECK_EQ(DataType<DType>::kFlag, DataType<SrcDType>::kFlag) << "Copy:type mismatch";`
This is just to get around compiler since partial specialization of functions are not allowed. This branch will never be executed with different type.